docs-go/flows.md

```diff
@@ -0,0 +1,187 @@
+# Flows
+
+Flows are functions with some additional characteristics: they are strongly
+typed, streamable, locally and remotely callable, and fully observable.
+Firebase Genkit provides CLI and developer UI tooling for working with flows
+(running, debugging).
+
+## Defining flows
+
+A flow wraps a function:
```

Comment:
In JS, this is correct: they are both a callable function and they wrap a function that does user-defined work. In Go, this is not strictly true. It's a struct that contains a function. But the concepts are the same so I wonder if we should stick with "flows = functions" because that's what we aim them to be, but because Go isn't as flexible as JS, to achieve the side effects of tracing, etc, we need to store them as structs.